### ISSUE TITLE
focused page breaks when firstDay is changed (3.0.3)

### ISSUE THREAD

**Issue by micimize** [aleksanderwozniak/table_calendar#631]
If I dynamically set `firstDay`, the display gets out of sync with the date inclusion calculations based on `_focusedDay`. Looking briefly into the source, I believe this could be solved by adjusting `_previousIndex` in `didUpdateWidget` of `TableCalendarBase`, but there may be more to it:
```dart
    _previousIndex = _calculateFocusedPage(
        widget.calendarFormat, widget.firstDay, _focusedDay);
```
I think the issue stems from `_calculateFocusedPage` doing it's calculations based on the cached `_previousIndex` + fresh `widget.firstDay`.

**Comment by aleksanderwozniak** [comment]
That's correct, `firstDay` and `lastDay` were originally intended to be static properties, as I believe that covers most of the use cases. I did however make a quick push - you will find it in `dynamic-bounds` branch. This still needs to be properly tested, as there might be some hard to spot edge case scenarios, but feel free to check it out. 

Bear in mind that you might have to update the `focusedDay` alongside `firstDay`/`lastDay`, as to not let it slip outside of the updated bounds.

**Comment by micimize** [comment]
Thanks! I gave it a quick test using my affected code and it seems to solve the issue.

Worth noting that I ended up making the design decision to start the calendar this particular component (nested calendar view per-item) at the start of all possible ranges for now, so I had to revert that change to test. Which is just to say, not a super-super robust user test and not an urgent change for me anymore.

https://user-images.githubusercontent.com/8343799/146651871-9b615e07-87e0-414d-9380-3e0a32233c08.mov
